ksqldb-functional-tests/src/test/resources/query-validation-tests/row-id.json

```diff
@@ -0,0 +1,227 @@
+      ],
+      "outputs": [
+        {"topic":  "OUTPUT", "key": "z", "value": {"foo": "a", "bar": "c", "rib": "eg=="}, "timestamp": 7},
+        {"topic":  "OUTPUT", "key": "z", "value": {"foo":  "b", "bar": "c", "rib": "eg=="}, "timestamp": 7}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "RIA BYTES KEY, FOO STRING, BAR STRING, RIB BYTES"}
+        ]
+      }
+    },
+    {
+      "name": "CAST ROWID",
+      "properties": {
+        "ksql.rowid.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM input (id STRING KEY, foo STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM output AS SELECT id, CAST(ROWID AS STRING) AS ri FROM input;"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "Invalid Select: Cast of BYTES to STRING is not supported."
+      }
+    },
+    {
+      "name": "Should fail if select ROWID without aliases",
+      "properties": {
+        "ksql.rowid.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM input (id INT KEY, foo STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM output AS SELECT id, ROWID FROM input;"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "Reserved column name in select: `ROWID`. Please remove or alias the column."
+      }
+    },
+    {
+      "name": "Filter using ROWID",
+      "properties": {
+        "ksql.rowid.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM TEST (ID INT KEY, a BYTES, b BYTES) WITH (kafka_topic='test', value_format='JSON');",
+        "CREATE STREAM TEST2 AS SELECT ID, ROWID AS RESULT FROM TEST WHERE ROWID BETWEEN a AND b;"
+      ],
+      "inputs": [
+        {"topic": "test","key": 0, "value": {"A": "AAAAAQ==", "B": "AAAAAQ=="}},
+        {"topic": "test","key": 1, "value": {"A": "AAAAAQ==", "B": "AAAAAQ=="}},
+        {"topic": "test","key": 2, "value": {"A": "AAAAAQ==", "B": "AAAAAg=="}},
+        {"topic": "test","key": 3, "value": {"A": "AAAAAQ==", "B": "AAAAAg=="}},
+        {"topic": "test","key": 3, "value": {"A": "AAAAAQ==", "B": "AAAAAw=="}},
+        {"topic": "test","key": 4, "value": {"A": "AAAAAQ==", "B": "AAAAAw=="}}
+      ],
+      "outputs": [
+        {"topic": "TEST2", "key": 1,"value": {"RESULT": "AAAAAQ=="}},
+        {"topic": "TEST2", "key": 2,"value": {"RESULT": "AAAAAg=="}},
+        {"topic": "TEST2", "key": 3,"value": {"RESULT": "AAAAAw=="}}
+      ]
+    },
+    {
+      "name": "GROUP BY ROWID",
+      "properties": {
+        "ksql.rowid.enabled": true
+      },
+      "statements": [
+        "CREATE STREAM L (A STRING KEY, B INT) WITH (kafka_topic='LEFT', value_format='JSON');",
+        "CREATE TABLE OUTPUT AS SELECT COUNT(B) AS bc, ROWID AS ri FROM L GROUP BY ROWID;"
```

Comment:
Same comment as above regarding repartitioning. This is currently creating a repartition topic, which is wasteful.